go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfunctions
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env/actions"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle"
+)
+
+const DoltCommitFuncName = "dolt_commit"
+
+type DoltCommitFunc struct {
+	children []sql.Expression
+}
+
+// NewDoltCommitFunc creates a new DoltCommitFunc expression.
+func NewDoltCommitFunc(args ...sql.Expression) (sql.Expression, error) {
+	return &DoltCommitFunc{children: args}, nil
+}
+
+// Trims the double quotes for the param.
+func trimQuotes(param string) string {
+	if len(param) > 0 && param[0] == '"' {
+		param = param[1:]
+	}
+
+	if len(param) > 0 && param[len(param)-1] == '"' {
+		param = param[:len(param)-1]
+	}
+
+	return param
+}
+
+func (d DoltCommitFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	// Get the information for the sql context.
+	dbName := ctx.GetCurrentDatabase()
+	dSess := sqle.DSessFromSess(ctx.Session)
+
+	ddb, ok := dSess.GetDoltDB(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load %s", dbName)
+	}
+
+	rsr, ok := dSess.GetDoltDBRepoStateReader(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load the %s RepoStateReader", dbName)
+	}
+
+	rsw, ok := dSess.GetDoltDBRepoStateWriter(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load the %s RepoStateWriter", dbName)
+	}
+
+	ap := actions.CreateCommitArgParser()
+
+	// Get the args for DOLT_COMMIT.
+	args := make([]string, 0)
```

Comment:
If you're building an array with append() calls like this in go, then it's more idiomatic to declare the slice like:

`var args []string`

On the other hand, if you know its size in advance, like you do here, then it's usually better (and more performant) to initialize it like:

`args := make([]string, len(d.children))`

Then assign to it with the index:

`args[i] = ...`

Instead of using append.

Either is fine, but pick one.